### DISCUSSION TITLE
SearchLogic with localised dates

### DISCUSSION THREAD

**Discussion by diereysaa** [Laravel-Backpack/community-forum#1175] [Q&A (Help)] [answered]
Hi Y'all,
I'm struggling to use the search on a column that has the `date` type. The column is localised, so its format is `DD MM YYYY` Take a look at it:
![image](https://github.com/user-attachments/assets/2d6449a1-bd01-41d8-9d61-dda1dc757dac)
Basically I want to offer the option to search on the "Inicio del curso" column, so when the user searches for "nov" it will show all the results that include that substring.

I've already tried several things:

```
        CRUD::column('start_date')
            ->label('Inicio del curso')
            ->type('date')
            ->searchLogic('text');
``` 
(Theoretically this should allow searching by the content as if it were just plain text, does not work :( )

---

```
        CRUD::column('start_date')
            ->label('Inicio del curso')
            ->type('date')
            ->searchLogic(function ($query, $column, $searchTerm) {
                $query->orWhere('start_date', 'like', '%'.$searchTerm.'%');
            });
```
(Can't make it work that way, because on the database, the field is `DATETIME` type, and searching for "nov" gives no result)

---

I'm pretty sure I'm missing something but I was attempting things for a few hours now, and couldn't find a solution.

Thanks in advance for all your comments.

Cheers!

**Comment by jcastroa87** [comment, marked as answer]
Hello @diereysaa 

If you are using MySQL, did you try with "DATE_FORMAT"

```
CRUD::column('start_date')
    ->label('Inicio del curso')
    ->type('date')
    ->searchLogic(function ($query, $column, $searchTerm) {
        $query->orWhereRaw("DATE_FORMAT(start_date, '%d %b %Y') LIKE ?", ['%' . $searchTerm . '%']);
    });
```

Cheers.

**Comment by diereysaa** [comment]
HUGE thanks Jorge!!! It worked like a charm 😆 

Thanks for the prompt response.
